src/plugins/intel_cpu/src/config.cpp

```diff
@@ -73,192 +68,248 @@ void Config::applyDebugCapsProperties() {
+            }
+        } else if (key == ov::hint::num_requests.name()) {
+            try {
+                ov::Any value = val.as<std::string>();
+                int val_i = value.as<int>();
+                if (val_i < 0)
+                    OPENVINO_THROW("invalid value.");
+                hintNumRequests = static_cast<uint32_t>(val_i);
+            } catch (const ov::Exception&) {
+                OPENVINO_THROW("Wrong value ",
+                               val.as<std::string>(),
+                               "for property key ",
+                               ov::hint::num_requests.name(),
+                               ". Expected only >0.");
+            }
         } else if (key == ov::hint::enable_cpu_pinning.name()) {
-            if (val == InferenceEngine::PluginConfigParams::YES) {
-                enableCpuPinning = true;
-                changedCpuPinning = true;
-            } else if (val == InferenceEngine::PluginConfigParams::NO) {
-                enableCpuPinning = false;
+            try {
+                enableCpuPinning = val.as<bool>();
                 changedCpuPinning = true;
-            } else {
-                IE_THROW() << "Wrong value " << val << "for property key " << ov::hint::enable_cpu_pinning.name()
-                           << ". Expected only true/false." << std::endl;
+            } catch (ov::Exception&) {
+                OPENVINO_THROW("Wrong value ",
+                               val.as<std::string>(),
+                               "for property key ",
+                               ov::hint::enable_cpu_pinning.name(),
+                               ". Expected only true/false.");
             }
         } else if (key == ov::hint::scheduling_core_type.name()) {
-            const auto core_type = ov::util::from_string(val, ov::hint::scheduling_core_type);
-            if (core_type == ov::hint::SchedulingCoreType::ANY_CORE ||
-                core_type == ov::hint::SchedulingCoreType::PCORE_ONLY ||
-                core_type == ov::hint::SchedulingCoreType::ECORE_ONLY) {
-                schedulingCoreType = core_type;
-            } else {
-                IE_THROW() << "Wrong value " << val << "for property key " << ov::hint::scheduling_core_type.name()
-                           << ". Expected only " << ov::hint::SchedulingCoreType::ANY_CORE << "/"
-                           << ov::hint::SchedulingCoreType::PCORE_ONLY << "/"
-                           << ov::hint::SchedulingCoreType::ECORE_ONLY << std::endl;
+            try {
+                schedulingCoreType = val.as<ov::hint::SchedulingCoreType>();
+            } catch (ov::Exception&) {
+                OPENVINO_THROW("Wrong value ",
+                               val.as<std::string>(),
+                               "for property key ",
+                               ov::hint::scheduling_core_type.name(),
+                               ". Expected only ANY_CORE/PCORE_ONLY/ECORE_ONLY");
```

Comment:
It is better to output full value like `ov::hint::SchedulingCoreType::ANY_CORE` in message. Then user can use them directly as input parameters.

Reply:
done